compaction/compaction_manager.hh

```diff
@@ -393,7 +394,7 @@ public:
     future<> perform_cleanup(replica::database& db, compaction::table_state& t);
 
     // Submit a table to be upgraded and wait for its termination.
-    future<> perform_sstable_upgrade(replica::database& db, replica::table* t, bool exclude_current_version);
+    future<> perform_sstable_upgrade(replica::database& db, compaction::table_state& t, sstables::sstable_version_types last_version, bool exclude_current_version);
```

Comment:
I thinks `s/last_version/version/` would be more fitting.

Reply:
We should be able to get to the table back from the table_state and through it get_highest_supported_format().
Adding it as a parameter is out of scope.
Also we may consider having a sstable instance per table_state / compaction_group as creation and deletion of sstables in it should be autonomous and therefore independent of the table.
(will become possibly once we switch to using uuid as the sstable generation - Cc @cmm)

Reply:
I don't understand why we're adding a parameter.

Reply:
> We should be able to get to the table back from the table_state and through it get_highest_supported_format(). Adding it as a parameter is out of scope. Also we may consider having a sstable instance per table_state / compaction_group as creation and deletion of sstables in it should be autonomous and therefore independent of the table. (will become possibly once we switch to using uuid as the sstable generation - Cc @cmm)

The idea of table_state is never referencing replica::table. But abstract table concept. table_state can be implemented without a replica::table at all, like for unit tests. 

I added a new parameter to avoid adding get_sstables_manager() to table_state interface, but I think it's perhaps better to just pass get_sstables_manager() to perform_sstable_upgrade().